### PR TITLE
POL-878: Azure Rightsize Compute Instances Memory Support and General Improvements

### PR DESCRIPTION
**Pull request by XOmniverse**
### Description

This adds support for memory metrics and brings over a bunch of the improvements being made for the AWS version of this policy to the Azure version.

Azure now supports memory usage ("Available Memory Bytes") via the Insights API, so this data is straightforwardly pulled from the same API that CPU data is pulled from. The percentage is calculated by finding the total memory of the instance's SKU via the Microsoft.Compute/skus API and then doing the appropriate calculations.

From the CHANGELOG:

- Several parameters altered to be more descriptive and human-readable
- Removed deprecated "Log to CM Audit Entries" parameter
- Added ability to only report recommendations that meet a minimum savings threshold
- Added support for memory metrics along with relevant parameters
- Added ability to configure how many days to consider CPU/memory statistics for
- Added ability to filter resources by multiple tag key:value pairs
- Added ability to make recommendations based on maximum CPU/memory usage
- Added additional context to incident description
- Normalized incident export to be consistent with other policies
- Added human-readable recommendation to incident export
- Policy no longer raises new escalations if statistics or savings data changed but nothing else has
- Streamlined code for better readability and faster execution

### Link to Example Applied Policy

https://app.flexera.com/orgs/6/automation/applied-policies/projects/7954?policyId=64b1457a12688300016785e1

(Note: Savings will show 0 for the above example due to the test environment. I've confirmed independently that the savings field works as expected in a normal environment)

### Contribution Check List

- [X] New functionality includes testing.
- [X] New functionality has been documented in the README if applicable
- [X] New functionality has been documented in CHANGELOG.MD

### CODE REVIEW

File: cost/azure/rightsize_compute_instances/README.md [hunk 1]
```diff
@@ -47,7 +52,11 @@ For administrators [creating and managing credentials](https://docs.flexera.com/
 
 - [**Azure Resource Manager Credential**](https://docs.flexera.com/flexera/EN/Automation/ProviderCredentials.htm#automationadmin_109256743_1124668) (*provider=azure_rm*) which has the following permissions:
   - `Microsoft.Compute/virtualMachines/read`
-  - `Microsoft.Compute/virtualMachines/write`
+  - `Microsoft.Compute/virtualMachines/write`*
+  - `Microsoft.Compute/skus/read`
+  - `Microsoft.Insights/Metrics/Read`
```
Comment (nia-vf1) [inline cost/azure/rightsize_compute_instances/README.md:57]: Small thing but

```suggestion
  - `Microsoft.Insights/metrics/read`
```

File: cost/azure/rightsize_compute_instances/azure_compute_rightsizing.pt [hunk 2]
```diff
@@ -20,102 +20,156 @@ info(
...
   description "Select the Azure Resource Manager Credential from the list."
   tags "provider=azure_rm"
 end
 
-#AUTHENTICATE WITH RIGHTSCALE/OPTIMA
 credentials "auth_flexera" do
   schemes "oauth2"
-  label "Flexera_Automation"
-  description "Select FlexeraOne OAuth2 credentials"
+  label "Flexera"
```
Comment (nia-vf1) [inline cost/azure/rightsize_compute_instances/azure_compute_rightsizing.pt:163]: should Flexera be lowercase f like `label "flexera"`? Think that's how it is for other policies (or at least most of them)

Comment (XOmniverse) [reply cost/azure/rightsize_compute_instances/azure_compute_rightsizing.pt:163]: No. It just determines how it appears in the UI, where capitalization looks a lot more professional.

Most of our policies were written like 5 years ago by people not too worried about things like presentability and UI consistency.

**Review by nia-vf1 — APPROVED** [review]
LGTM

**Pull request merged**